packages/dd-trace/src/collector/exporter.js

```diff
@@ -0,0 +1,40 @@
+'use strict'
+
+const { channel } = require('dc-polyfill')
+const { Encoder } = require('./encoder')
+
+const segmentStartChannel = channel('datadog:tracing:segment:start')
+const segmentDiscardChannel = channel('datadog:tracing:segment:discard')
+const spanStartChannel = channel('datadog:tracing:span:start')
+const spanFinishChannel = channel('datadog:tracing:span:finish')
+const tagsChannel = channel('datadog:tracing:span:tags')
+const errorChannel = channel('datadog:tracing:span:error')
+
+class CollectorExporter {
+  constructor (config) {
+    const encoder = this._encoder = new Encoder(config)
+
+    this._handlers = new Map([
+      [segmentStartChannel, encoder.encodeSegmentStart.bind(encoder)],
+      [segmentDiscardChannel, encoder.encodeSegmentDiscard.bind(encoder)],
+      [spanStartChannel, encoder.encodeSpanStart.bind(encoder)],
+      [spanFinishChannel, encoder.encodeSpanFinish.bind(encoder)],
+      [tagsChannel, encoder.encodeAddTags.bind(encoder)],
+      [errorChannel, encoder.encodeException.bind(encoder)]
+    ])
+  }
+
+  start () {
+    this._handlers.forEach((cb, ch) => ch.subscribe(cb))
```

Comment:
nitpick: cb and ch are difficult to tell apart when reading fast

Reply:
Is it fine to leave it as-is? While I understand the sentiment, it's just a very simple one-liner, so I prioritize keeping the line short when possible.